h2o-parsers/h2o-parquet-parser/src/main/java/water/parser/parquet/ParquetParser.java

```diff
@@ -239,9 +244,11 @@ private static ParquetPreviewParseWriter readFirstRecords(ParquetParseSetup init
   }
 
   private static BlockMetaData findFirstBlock(ParquetMetadata metadata) {
+    if (metadata.getBlocks().isEmpty()) return null;
+
     BlockMetaData firstBlockMeta = metadata.getBlocks().get(0);
     for (BlockMetaData meta : metadata.getBlocks()) {
-      if (firstBlockMeta.getStartingPos() < firstBlockMeta.getStartingPos()) {
```

Comment:
Great catch! This could be source of some unexplained issues!

Reply:
May I squash&merge ?

Reply:
Is it working for multifile when one of the files is empty?

Reply:
Tested on two files, one with headers and one with data. Works.